dagql/call/callpbv1/call.proto

```diff
@@ -80,6 +80,12 @@ message Call {
   // A unique digest of this Call. Note that this must not be set when
   // calculating the Call's digest.
   string digest = 9;
+  
+  // The view that this call was made in. Since a graphql server may present
+  // slightly different views depending on the specified view, to be able to
+  // evaluate calls wherever, we need to track which view the call was actually
+  // called in.
+  string view = 10;
```

Comment:
One thought I just had, since this ends up in IDs it will impact the caching of function calls (once that's in place). Right? 

If so, while It's a bit complicated to think about, I *think* that's sometimes exactly what we'd want and other times not ideal.

E.g. if you received an ID as an arg from a module on an older engine version that involves back-incompat changes, it probably makes sense the ID would result in a different cache key (I think...).
* *But* if there's modules on say v0.12.0 and v0.12.1 and there were no breaking changes there, then the different versions would result in different cache keys for the function call unnecessarily

Not even 100% sure yet this is true and even if it is, I can't tell how big an issue it would be or the best way to fix it yet. 

So not a blocker, just calling out in case others have more thoughts.

Reply:
Mm, I hadn't thought about this :thinking: But yes, this would break cache.

Suppose we had two modules build containers, but one is doing so with schema X, and the other does so with schema Y. Even if they do exactly the same thing, they're gonna have different IDs - which would impact module caching (but wouldn't impact any `WithExec` caching, since the underlying LLB should be the same). I do think we want the IDd behavior for safety, it's not safe to just always alias these as you point out.

Aside: frustratingly, if we wanted to avoid this behavior it would be a total pain - there's really not anywhere else we can attach this info. There needs to be someway to link IDs to where they were constructed.

One option that might not suck - if a field is a member of the default global view (no view has been defined for it), then we can set the ID's view to the empty string `""`. Then every view of the server will have the same IDs for those fields, *but* we'd need to undo this logic (which requires a bit more explicitness in schema defs, but not too bad): https://github.com/dagger/dagger/blob/12ca11a58c3d84275b5b8d04cd17e1735a3b4fa5/dagql/objects.go#L95-L97

We'd need to undo that, to actually *ensure* that every view into the schema always gets the same results for items in the global view.

At least this way, only the incompatible APIs would break cache (exactly what we want). Will give this a go to see if this is at all feasible.

Reply:
Added in [`b092e39` (#7759)](https://github.com/dagger/dagger/pull/7759/commits/b092e39d3227e0f2e98196913a8eef6b637049c7).